.github/workflows/build_and_test.yml

```diff
@@ -180,13 +180,13 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        group: [1, 2, 3, 4, 5, 6]
+        group: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
```

Comment:
This might cause jobs to get queued more easily and therefore increase time.

I think we have limited concurrency but lets check with @ijjk first

Reply:
We should have autoscaling now, but not sure if there is an additional concurrency limit.

I adjusted the jobs to be ~10min per job for each...